prov/usnic/src/usdf_fabric.c

```diff
@@ -189,6 +190,42 @@ usdf_fill_addr_info(struct fi_info *fi, uint32_t addr_format,
 	return ret;		// fi_freeinfo() in caller frees all
 }
 
+void build_devname(struct usd_device_attrs *dap, char **bufp)
```

Comment:
My $0.02: if you have a function named `build_devname()`, that implies to me that it's going to build the devname for *all* versions.  I.e., you shouldn't only call this function for >=v1.4.

Specifically: it might be a little better down below to *always* call this function (and maybe name this function `get_getname()`?), and then have the version comparison / behavior difference in this function.